File: app/src/main/java/com/privat/contacts/domain/model/AddressDomain.java
package com.privat.contacts.domain.model;

import org.jetbrains.annotations.NotNull;

public interface AddressDomain {
    @NotNull
    <T> T map(Mapper<T> mapper);

    interface Mapper<T> {
        @NotNull
        T map(
                int userId,
                String city,
                String streetName,
                String streetAddress,
                String zipCode,
                String state,
                String country,
                double coordinateLat,
                double coordinateLng);
    }
}

File: app/src/main/java/com/privat/contacts/domain/model/AddressDomain.kt
package com.privat.contacts.domain.model

interface AddressDomain {
    fun <T> map(mapper: Mapper<T>): T
    interface Mapper<T> {
        fun map(
                userId: Int,
                city: String,
                streetName: String,
                streetAddress: String,
                zipCode: String,
                state: String,
                country: String,
                coordinateLat: Double,
                coordinateLng: Double): T
    }
}
